cmd/flux/install.go

```diff
@@ -183,6 +183,20 @@ func installCmdRun(cmd *cobra.Command, args []string) error {
 	logger.Successf("manifests build completed")
 	logger.Actionf("installing components in %s namespace", *kubeconfigArgs.Namespace)
 
+	kubeClient, err := utils.KubeClient(kubeconfigArgs, kubeclientOptions)
+	if err != nil {
+		return err
+	}
+
+	info, installed, err := getFluxClusterInfo(ctx, kubeClient)
+	if err != nil {
+		return fmt.Errorf("error checking for exisiting flux installation: %w", err)
+	}
+
+	if installed && info.bootstrapped {
+		return fmt.Errorf("this cluster has already been bootstrapped with Flux %s! Please use 'flux bootstrap' to upgrade", info.version)
+	}
```

Comment:
I wonder if any false positives could arise from this new behaviour in certain edge cases. Maybe a `--force` flag could help override the check in these cases.

Reply:
I don't think we should allow people to force apply if the CRD is managed by Flux. The should uninstall and start fresh, forcing will only do harm as kustomize-controller will undo it.